### PR TITLE
Feature/578

### PR DESCRIPTION
**Pull request by MMashivskij**
## Description

Add "SimilarProducts" component to "Cart" component. 
Add algorythm to filter products as similar.

#### Screenshots

Original             |  Updated
:-------------------------:|:-------------------------:
Cart_before  | Cart_after

### Checklist
- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date admin and back-end part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

**Review by dmytro07 — APPROVED** [review]

**Review by IlyaBelik — DISMISSED** [review]

**Review by dmytro07 — DISMISSED** [review]

**Comment by Haytham1337** [comment]
When I go to another product page, the similar products block should not display the old values ​​from the previous product.
image

File: src/components/number-input/number-input.js [hunk 1]
```diff
@@ -17,11 +21,15 @@ const NumberInput = ({ onChangeQuantity, quantity }) => {
...
-      <h3>{quantity}</h3>
-      <Button
-        className={styles.button}
-        onClick={() => onChangeQuantity(quantity + 1)}
-      >
+      <TextField
+        type='number'
+        value={quantity}
+        id='filled-basic'
+        variant='outlined'
```
Comment (IlyaBelik) [inline src/components/number-input/number-input.js:28]: please don`use magic strings

File: src/containers/orders/cart/cart-item/cart-item.js [hunk 2]
```diff
@@ -30,40 +37,42 @@ const CartItem = ({ item, language, currency, calcPrice, isCartEditing }) => {
 
   return (
     <TableRow classes={{ root: styles.root }} data-cy='cart-item'>
-      <TableCell classes={{ root: styles.image }} data-cy='cart-item-img'>
-        <Link to={`/product/${item._id}`}>
-          <b />
+      <TableCell data-cy='cart-item-img'>
```
Comment (IlyaBelik) [inline src/containers/orders/cart/cart-item/cart-item.js:40]: same here

File: src/containers/orders/cart/filled-cart/filled-cart.js [hunk 3]
```diff
@@ -1,23 +1,35 @@
...
+      cartList: Cart.list,
+      cartLoading: Cart.loading,
+      cartUserTotalPrice: Cart.totalPrice,
+      user: User.userData
+    })
   );
 
+  const totalPrice = items.reduce((acc, item) => acc + calcPriceForCart(item, currency), 0);
+  const currencySign = selectCurrencySign(currency, faHryvnia, faDollarSign);
+  console.log(cartLoading);
```
Comment (IlyaBelik) [inline src/containers/orders/cart/filled-cart/filled-cart.js:29]: remove console.log

**Review by Haytham1337 — APPROVED** [review]

**Review by IlyaBelik — APPROVED** [review]

**Review by dmytro07 — APPROVED** [review]

**Pull request merged**